ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/AuthHandlers.java

```diff
@@ -58,28 +68,36 @@ static void setupAuthHandlers(final Server server, final Router router,
     systemAuthenticationHandler.ifPresent(handler -> registerAuthHandler(router, handler));
 
     if (jaas.isPresent() || authenticationPlugin.isPresent()) {
+      final List<String> skipPaths = server.getConfig()
+              .getList(KsqlRestConfig.AUTHENTICATION_SKIP_PATHS_CONFIG);
+      final Pattern skipPathPattern = getAuthenticationSkipPathPattern(skipPaths);
+
       registerAuthHandler(
           router,
-          rc -> selectHandler(rc, jaas.isPresent(), pluginHandler.isPresent()));
+          rc -> selectHandler(rc, skipPathPattern, jaas.isPresent(), pluginHandler.isPresent()));
 
       // set up using JAAS
-      jaas.ifPresent(h -> registerAuthHandler(router, wrappedHandler(h, Provider.JAAS)));
+      jaas.ifPresent(h -> registerAuthHandler(router, selectiveHandler(h, Provider.JAAS::equals)));
 
       registerAuthHandler(
           router,
-          wrappedHandler(new RoleBasedAuthZHandler(
+          selectiveHandler(new RoleBasedAuthZHandler(
               server.getConfig().getList(KsqlRestConfig.AUTHENTICATION_ROLES_CONFIG)),
-              Provider.JAAS));
+              Provider.JAAS::equals));
 
       // set up using PLUGIN
-      pluginHandler.ifPresent(h -> registerAuthHandler(router, wrappedHandler(h, Provider.PLUGIN)));
+      pluginHandler.ifPresent(h -> registerAuthHandler(router,
+              selectiveHandler(h, Provider.PLUGIN::equals)));
 
       // For authorization use auth provider configured via security extension (if any)
       securityExtension.getAuthorizationProvider()
           .ifPresent(ksqlAuthorizationProvider ->
               registerAuthHandler(
                   router,
-                  new KsqlAuthorizationProviderHandler(server, ksqlAuthorizationProvider)));
+                  selectiveHandler(
+                          new KsqlAuthorizationProviderHandler(server, ksqlAuthorizationProvider),
+                          provider -> provider == Provider.JAAS || provider == Provider.PLUGIN
```

Comment:
Did we just implicitly only fall through to here if it was jaas or plugin and now you're just formally checking that?

Reply:
That makes sense.  It's better to have it centralized here.